.travis.yml

```diff
@@ -1,38 +1,106 @@
-- "3.5"
-- "3.6-dev"
-- "3.7-dev"
-- "nightly"
 
 dist: trusty
 sudo: false
 cache: pip
 
 before_install:
 - pip install --upgrade flit
-install:
-- pushd blurb
-- flit install
-- popd
-script:
-- blurb test
 
 .mixtures:
+- &run-if-tagged
+  if: tag IS present
+- &run-if-cherry-picker
+  if: tag =~ ^cherry\-picker\-v\d+\.\d+\.\d+$
+- &run-if-blurb
+  if: tag =~ ^blurb\-v\d+\.\d+\.\d+$
+- &run-if-cherry-picker-or-untagged
+  if: tag IS NOT present OR tag =~ ^cherry\-picker\-v\d+\.\d+\.\d+$
+- &run-if-blurb-or-untagged
+  if: tag IS NOT present OR tag =~ ^blurb\-v\d+\.\d+\.\d+$
 - &install-and-test-cherry-picker
+  <<: *run-if-cherry-picker-or-untagged
+  env:
+    TARGET_PKG: cherry_picker
   install:
-  - pushd cherry_picker
+  - &cd-to-project pushd "$TARGET_PKG"
   - flit install
-  - popd
   script:
-  - pushd cherry_picker
   - pytest cherry_picker/test.py -v
   - popd
+- &install-and-test-blurb
+  <<: *run-if-blurb-or-untagged
+  env:
+    TARGET_PKG: blurb
+  install:
+  - *cd-to-project
+  - flit install
+  - popd
+  script:
+  - blurb test
+- &deploy-base
+  stage: Publish dists to PYPI
+  <<: *run-if-tagged
+  python: "3.6-dev"
+  install:
+  - *cd-to-project
+  script:
+  - flit build
+  before_deploy:
+  # Add an empty setup.py stub, because pypi provider always calls it
+  - touch setup.py
+  deploy: &deployment-config
+    provider: pypi
+    # `skip-cleanup: true` is required to preserve binary wheel and sdist,
+    # built by during `install` step above.
+    skip-cleanup: true
+    # `skip-existing: true` is required to skip uploading dists, already
+    # present in PYPI instead of failing the whole process.
+    # This happenes when other CI (AppVeyor etc.) has already uploaded
+    # the very same dist (usually sdist).
+    skip-existing: true
+    user: &pypi-user Mariatta  # TODO: consider having a separate "uploader user"
+                               # with reduced priviliges for security reasons
+    password: &pypi-password
+      # Encrypt with `travis encrypt -r python/core-workflow --org` while using travis-ci.org;
+      # change to `travis encrypt -r python/core-workflow --api-endpoint 'https://api.travis-ci.com/'`
+      # upon switch to __free__ travis-ci.com:
+      secure: PLACE_YOUR_ENCRYPTED_PASSWORD_HERE
+    on:
+      tags: true
+      all_branches: true
+
 jobs:
   include:
+  - python: "3.5"
+    <<: *install-and-test-blurb
+  - python: "3.6-dev"
+    <<: *install-and-test-blurb
+  - python: "3.7-dev"
+    <<: *install-and-test-blurb
+  - python: "nightly"
+    <<: *install-and-test-blurb
+
   - python: "3.6-dev"
     <<: *install-and-test-cherry-picker
   - python: "3.7-dev"
     <<: *install-and-test-cherry-picker
   - python: "nightly"
     <<: *install-and-test-cherry-picker
+
+  - <<: *deploy-base
+    <<: *run-if-cherry-picker
+    env:
+      TARGET_PKG: cherry_picker
+
+  - <<: *deploy-base
+    <<: *run-if-blurb
+    env:
+      TARGET_PKG: blurb
+    deploy:
+      <<: *deployment-config
```

Comment:
I don't know if blurb maintainers (I'm not one of them) want the same capability or not. Maybe we need to remove this portion.

Reply:
I don't really like removing stuff. Instead, we can add smth like`if: false` to disable this. Deal?

Reply:
Ok :)